### ISSUE TITLE
Allow bulk CModel changes

### ISSUE THREAD

**Issue by bondjimbond** [mjordan/islandora_change_cmodel#6]
In the Collection management tab, it would be helpful to add an additional option: "Change child Cmodels". Like the migrate/delete/share options, generate a select list of collection child objects and run a cmodel change in a batch.

**Comment by mjordan** [comment]
@bondjimbond currently the code that changes the content model is in a [function](https://github.com/mjordan/islandora_change_cmodel/blob/7.x/includes/utilities.inc#L17) that requires a `$form_state` as a parameter, but it would make sense to break out that code into its own function so that other modules could use it. I'll do that.

But, I probably won't have time this summer to add a batch as you describe. Would be a useful feature, but a very dangerous one. Would probably need to include at least an option to autoregenerate derivatives, correct?

**Comment by mjordan** [comment]
Opened #7 to make that code more portable.

**Comment by bondjimbond** [comment]
PRETTY dangerous, but I'm not really sure how bad it could be. Maybe not so bad if there was a log kept with an option to revert?

> Would probably need to include at least an option to autoregenerate derivatives, correct?

It would be nice, not strictly necessary but certainly helpful for a batch function. Perhaps it could hook into islandora_batch_derivative_trigger? (https://github.com/qadan/islandora_batch_derivative_trigger)

**Comment by mjordan** [comment]
Yes, pretty dangerous. I'm honestly not too worried about the "danger" of the ability to change an object's cmodel, but as soon as you can do it en bulc, any danger present is amplified. One thing to remember is that changing an object's cmodel changes its RELS-EXT datastream, and that might (but it's unlikely in practice I would say) trigger other hooks, resulting in a cascade effect.

An option to revert would be sweet. Begs the question of how long the ability to revert remains available, but that could be worked into this feature too.

**Comment by bondjimbond** [comment]
> An option to revert would be sweet. Begs the question of how long the ability to revert remains available, but that could be worked into this feature too.

I'm thinking of how Islandora Batch works - it creates sets, which remain saved until you delete them (unless you've set it to automatically delete).
